server/osc_dash.py

```
@@ -0,0 +1,402 @@
+    const candle = `<div class="candle-wrap"><div class="candle-bar"><div class="candle-wick" style="left:${wickLeft}%;width:${wickW}%;"></div><div class="candle-body" style="left:${bodyLeft}%;width:${Math.max(2,bodyW)}%;background:${bodyColor};border:1px solid ${bodyColor}"></div><div style="position:absolute;left:50%;top:0;bottom:0;width:1px;background:var(--faint);opacity:.6"></div></div><div style="display:flex;justify-content:space-between;font-size:8px;color:var(--faint);margin-top:2px"><span>1</span><span>50</span><span>99</span></div><div style="font-size:10px;color:var(--dim);margin-top:1px">טווח ${((h-l)).toFixed(1)}¢ · סגירה ${(c).toFixed(1)}¢</div></div>`;
+    tbl+=`<tr><td style="font-weight:700">${esc(w.label)}</td><td class="mono" title="${esc(w.slug)}" style="font-size:12px">${esc(w.slug.slice(-14))}<div style="font-size:10px;color:var(--faint)">${new Date(w.start_ts*1000).toLocaleTimeString('he-IL',{hour:'2-digit',minute:'2-digit'})}</div></td><td>${openHtml}</td><td>${upHtml}</td><td>${downHtml}</td><td>${candle}</td><td>${clsPill(w.class)}</td><td><a href="${w.url}" target="_blank" rel="noopener" style="font-size:13px;font-weight:700">פתח ↗</a></td></tr>`;
+  }
+  tbl+='</table><div class="note" style="font-size:12px"><b>התחיל ב</b> נמדד באמת — ה-mid הראשון שנדגם אחרי שהחלון נפתח (לכן לפעמים 47.5¢ ולא 50¢: החלון כבר רץ 10-20 שניות כשהתחלנו למדוד אותו). חלונות שייפתחו מעכשיו יתחילו קרוב ל-50¢. <b style="color:var(--up)">עלה ל</b> מודד מ-50: 50→60 = <span class="price-up">+10¢</span>, <b style="color:var(--down)">ירד ל</b> = 50→40 = <span class="price-down">−10¢</span>. הנר מראה פתיחה (קצה שמאלי של הגוף), סגירה (קצה ימין), ופתילות ל-high/low (בחרנו צד UP; DOWN הוא 100-UP).</div></div>';
+  $('tables').innerHTML=tbl;
+}
+tick(); setInterval(tick,3000);
+</script></body></html>
+"""
+PAGE_SUMMARY = r"""<!doctype html><html lang="he" dir="rtl"><head>
+<meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1">
+<title>סיכום סטטיסטי — 630 חלונות</title>
```

Ruff (0.16.2)

[warning] String contains ambiguous `ס` (HEBREW LETTER SAMEKH). Did you mean `o` (LATIN SMALL LETTER O)?

(RUF001)